logger.go

```diff
@@ -18,6 +15,10 @@ type Logger interface {
 	Error(err error, msg string, keysAndValues ...interface{})
 }
 
+// DefaultLogger is used if none is specified.
+func createDefaultLogger() Logger {
```

Comment:
In this way we will create multiple logger structs will lead to different mutexes in logger atomic methods